aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -366,14 +438,16 @@ impl<
     /// Read the latest value corresponding to a tag at a given group (identified by key).
     /// Return the size of the group (if requested), as defined above, alongside the version
     /// information (None if storage/pre-block version).
-    pub fn read_from_group(
+    /// If the layout of the resource is current UnSet, this function sets the layout of the
+    /// group to the provided layout.
+    pub fn fetch_tagged_data(
         &self,
         key: &K,
         tag: &T,
         txn_idx: TxnIndex,
-    ) -> anyhow::Result<(Version, Arc<V>, Option<Arc<MoveTypeLayout>>), MVGroupError> {
-        match self.group_values.get(key) {
-            Some(g) => g.get_latest_tagged_value(tag, txn_idx),
+    ) -> anyhow::Result<(Version, ValueWithLayout<V>), MVGroupError> {
+        match self.group_values.get_mut(key) {
```

Comment:
ditto: I don't see why mut here?

Reply:
Artifact of my previous design. Thanks for pointing it out. Fixed it.